### PR TITLE
feat(pipelines): CI/CD implementation 

### PR DESCRIPTION
**Pull request by muskan2622**
### Functionality:
CI/CD implementation to show workflow run

### Solution:
Check the Action workflow of the given github repository

## WtId:
<-- please enter your WorktezId -->

### Risk level:
- [x] high
- [ ] medium
- [ ] low

### How to test:
Go to pipelines section and check

### CODE REVIEW

**Review by sanjaykrishna1203 — APPROVED** [review]

File: src/app/services/team/team-service.service.ts [hunk 1]
```diff
@@ -93,6 +95,13 @@ export class TeamServiceService {
     })
   }
 
+
+  getCompletedRuns(owner: string, repo: string): Observable<GitDetails[]> {
+    const url = `${environment.githubApiUrl}/repos/${owner}/${repo}/actions/runs?status=completed`;
+    return this.httpClient.get<GitDetails[]>(url);
+  }
```
Comment (viv9k) [inline src/app/services/team/team-service.service.ts:102]: Better we move this to github service

**Review by viv9k — CHANGES_REQUESTED** [review]
Please  fix it

File: src/app/services/team/team-service.service.ts [hunk 2]
```diff
@@ -34,7 +36,7 @@ export class TeamServiceService {
   public gitDataReay: boolean = false;
   public labelsReady: boolean = false;
   public teamsReady: boolean = false;
-  constructor(private functions: AngularFireFunctions) { }
+  constructor(private functions: AngularFireFunctions, private httpClient: HttpClient) { }
```
Comment (viv9k) [inline src/app/services/team/team-service.service.ts:39]: There is no use of httpClient here

File: src/app/services/github-service/github-service.service.ts [hunk 3]
```diff
@@ -166,4 +169,16 @@ export class GithubServiceService {
...
   }
+
+  getCompletedWorkflowRuns(projectLink: string) {
+    const url = environment.githubApiUrl +"/repos/"+projectLink+"/actions/runs?status=completed";
+    return this.httpClient.get(url);
+  }
+
+
+  getCompletedRuns(owner: string, repo: string): Observable<GitDetails[]> {
+    const url = `${environment.githubApiUrl}/repos/${owner}/${repo}/actions/runs?status=completed`;
```
Comment (viv9k) [inline src/app/services/github-service/github-service.service.ts:180]: The url formula format can be simplyfied

**Review by viv9k — CHANGES_REQUESTED** [review]
Please fix it asap

File: src/app/services/github-service/github-service.service.ts (hunk 3 above)
Comment (simran142002) [reply src/app/services/github-service/github-service.service.ts:180]: please update your changes

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**